### DISCUSSION TITLE
1.3.7 is the last version where I can successfully generate a token

### DISCUSSION THREAD

**Discussion by top-gun** [llamafilm/tesla-http-proxy-addon#70] [General]
I have successfully enrolled my car with the tokens I generated with your add-on in version 1.3.7, and the switch to 2.1.1 broke the full process for me, as I am unable to get a working token from the "Auth for Tesla" app on my iPhone. 

Maybe I made a stupid mistake, but the tokens I create via your OAuth web process have a completely different format. What I get via your 1.3.7 add-on looks like 

EU_29d6f5c438cba11235948f986f4c16a1936c34414a352ffa25be3048614cxxxx

The tokens I get from the Auth for Tesla app after filling in my developer account detail and 2FA-password look like 

eyJhbGciOiJSUzI1NiIsInR5cCI6IkpXVCIsImtpZCI6IkpJcFJCQWlsQ0V1cXZRRjlwUHRtZkhKWnNjYyJ9.eyJpc3MiOiJodHRwczovL2F1dGgudGVzbGEuY29tL29hdXRoMi92MyIsInNjcCI6WyJvcGVuaWQiLCJvZmZsaW5lX2FjY2VzcyJdLCJhdWQiOiJodHRwczovL2F1dGgudGVzbGEuY29tL29hdXRoMi92My90b2tlbiIsInN1YiI6ImE1Njc3YWIxLTM0YjQtNDRhZS04MzNlLWVlZWM3MjhmNmFmMCIsImRhdGEiOnsidiI6IjEiLCJhdWQiOiJodHRwczovL293bmVyLWFwaS50ZXNsYW1vdG9ycy5jb20vIiwic3ViIjoiYTU2NzdhYjEtMzRiNC00NGFlLTgzM2UtZWVlYzcyOGY2YWYwIiwic2NwIjpbIm9wZW5pZCIsIm9mZmxpbmVfYWNjZXNzIl0sImF6cCI6Im93bmVyYXBpIiwiYW1yIjpbInB3ZCJdLCJhdXRoX3RpbWUiOjE3MTE3Mzk0NzN9LCJpYXQiOjE3MTE3Mzk0NzN9.MkdBIgdLv8LUHGY0WIlCmrwxXFaNIaBYANwP4yOLsWM25W86RBQbs-rqQfLxQuP_GofEgKAHqQiYio469hwjPNiohsB0u3LyyiDPygdtq_9a4zxP-VifJuZUZ36Vh3RgEQeXfWbZSFJeUYmTl2FYVtWu2ZJyNAVosu2JgxBfq-rJAlSzN3fJsX2Sca4PnAUthieJDUCFXgHUJHdxB8Gt1Il3u6gI9mA6P-9wioX3PXoDHeVY3asDQhDPd2FBuGaRbdKXIhtDQo_NhJEOdn1JhDH0lIil37Az_hcAVMwacOQp2BXEdAdTNPd-oKBFrcYClJeKz8ppwiIGd-lYAdbM_A

The short one from version 1.3.7 works perfectly fine in the Tesla integration, while the long one from the auth for Tesla app won't.

I guess I'm missing something obvious, but I have spent a full afternoon around this issue, so my only solution is to ask, sorry. My only solution was to restore my HA with the backup I made with version 1.3.7, and generate a token with the old workflow. That one worked immediately.

**Comment by llamafilm** [comment]
The Tesla Auth app supports both the old style and the new Fleet API token. It sounds like you’ve chosen the old style, or maybe you’re running an old version.

**Reply by top-gun** [reply]
Thank you for your fast reply. The app is the latest release for iOS, 2.0.0. I used the middle tab which is called Fleet API and after entering password and MFA, the flow ends with a short success message but I see no token. I guess it is related to the entry for the callback entry. I used https://xyz.duckdns.org/callback, just as I did for your plugin. I guess that‘s wrong. When I try https://xyz.duckdns.org, the apptells me the URI is not recognized and I use the uri found in my app address.

At this point, I‘m lost. How is this supposed to work? The old callback url seems not to work, and there is no guidance what to use instead.

**Reply by top-gun** [reply]
This is the full flow in screenshots


![IMG_0707](https://github.com/llamafilm/tesla-http-proxy-addon/assets/3148118/70cab63f-02fc-4efb-b6e4-dfacd37960dd)
![IMG_0708](https://github.com/llamafilm/tesla-http-proxy-addon/assets/3148118/22cbe9ae-652a-45c8-963c-46a6b1cd3434)
![IMG_0709](https://github.com/llamafilm/tesla-http-proxy-addon/assets/3148118/3b21df71-94ef-4ad6-9422-2324ce5781aa)
![IMG_0710](https://github.com/llamafilm/tesla-http-proxy-addon/assets/3148118/0089b7a2-32c9-442c-811d-97a65d8d36d4)
![IMG_0711](https://github.com/llamafilm/tesla-http-proxy-addon/assets/3148118/80bf8dd8-7bf7-402a-a15a-15ece365b0c7)

**Reply by llamafilm** [reply]
The callback URL must be the same one registered in your Tesla developer account. The benefit of using this app instead of my addon is that the callback URL doesn’t need to actually be functional, because this app is supposed to intercept the request and just extract the code. What is the short success message you got? If this app isn’t working, it’s probably best to open an issue on their GitHub. 
One thing I noticed is this app doesn’t offer a choice between NA and EU region. So that might be a problem.

**Reply by top-gun** [reply]
Yes, the url was the same - I just noticed that it‘s no longer reachable with 2.1.1, and thought this messed up in my setup.

Thank you for that valuable explanation. This helps me a lot with understanding the process. I will open an issue on their Github.

Fwiw, I just installed 2.2.0 and am very happy that a new process was added. This works just fine for me, and it will also help all the people who use Android, as it seems there is no token generator for Android with Fleet API capability.

**Reply by llamafilm** [reply]
Yes, I also noticed there is no good solution for Android, that's why I added it here.  I'm glad it works for you!
